### PR TITLE
Intro to Rust MOOC Section 5

### PR DESCRIPTION
**Pull request by CrackTheCode016**
- [x] Structs
- [x] Defining Methods for Structs
- [x] Vectors, Strings & Hashmaps
- [x] Vectors vs Strings - what’s the difference?

### CODE REVIEW

**Comment by CrackTheCode016** [comment]
All base content added - not terribly extensive, but enough for the module to be done.

File: docs/Rust/section5/collections.md [hunk 1]
```diff
@@ -3,4 +3,98 @@ id: collections
 title: Collections in Rust - Vectors, Strings, Hashmaps
 sidebar_label: Collections - Vectors, Strings, Hashmaps
 description: Learn what collections are, and when to use which in Rust.
----
\ No newline at end of file
+---
+
+We've already dealt with several collections.  Rust's standard library provides these data structures and aid in **dynamic** data manipulation and representation.  Similar arrays and tuples, collections can store more than value.  In contrast to such compound types, collections point to items stored on the **heap**.    The benefit of it being managed on the heap allows collections to be growable and generally more modifiable in runtime.  One collection you have already encountered numerous times throughout this course is `String`, which represents a list of characters.
```
Comment (DrW3RK) [inline docs/Rust/section5/collections.md:8]:
```suggestion
We've already dealt with several collections.  Rust's standard library provides these data structures and aid in **dynamic** data manipulation and representation.  Similar to arrays and tuples, collections can store more than value.  In contrast to such compound types, collections point to items stored on the **heap**.    The benefit of it being managed on the heap allows collections to be growable and generally more modifiable in runtime.  One collection you have already encountered numerous times throughout this course is `String`, which represents a list of characters.
```

File: docs/Rust/section5/collections.md [hunk 2]
```diff
@@ -3,4 +3,98 @@ id: collections
...
+
+There are three commonly used collections in Rust: 
+
+- Vectors - similar to arrays, where a list of items stored in chronological order is defined.
```
Comment (DrW3RK) [inline docs/Rust/section5/collections.md:12]: I am not sure if "chronological" is the right word here, as it implies the data in vectors somehow carries a timestamp or represents the passage of time somehow. You can correct me if I am wrong here.

```suggestion
- Vectors -  re-sizable arrays. Like slices, their size is not known at compile time, but they can grow or shrink at any time. 
```

File: docs/Rust/section5/collections.md [hunk 3]
```diff
@@ -3,4 +3,98 @@ id: collections
...
+let four: i32 = &vector[2];
+```
+
+A vector follows all rules of the borrow checker - meaning if the vector is mutable, then an immutable reference to an element is impossible.  Vectors also aren't limited to scalar types - you may define a struct, and store a list of structs within a vector if desired.
+
+## Strings
+
+The `String` collection should be very familiar by now.  `String` is a collection of characters stored on the heap.  The `&str` type, or string slice/literal is different from this type, as it's a primitive type, whereas `String` is a data structure part of the Rust standard library.  At its core, a String is a wrapped `Vec` of bytes.  
+
+Strings are rather complicated yet advantageous data structures.  If you want to learn more about Strings at a more technical level, please read the Rust book, as this is just meant to introduce how to use `String`.
```
Comment (DrW3RK) [inline docs/Rust/section5/collections.md:53]:
```suggestion
Strings are rather complicated yet advantageous data structures.  If you want to learn more about Strings at a more technical level, please read the [Rust book](https://doc.rust-lang.org/stable/book/), as this is just meant to introduce how to use `String`.
```

**Review by DrW3RK — APPROVED** [review]

**Pull request merged**